src/cocoa/toga_cocoa/widgets/webview.py

```diff
@@ -75,15 +76,15 @@ async def evaluate_javascript(self, javascript):
         :param javascript: The javascript expression to evaluate
         :type  javascript: ``str``
         """
-        return self.native.stringByEvaluatingJavaScriptFromString(javascript)
+        return self.native.evaluateJavaScript(javascript, completionHandler=None)
```

Comment:
I don't think this is quite correct - this should be returning the result of evaluating the javascript, which IIRC requires providing (and evaluating) the completion handler.

Reply:
Yes, I had overlooked this. In principle, I have managed to provide a completion handler and retrieve the result from it. The only issue: the result seems to be returned as a objc id. Is it possible to retrieve the corresponding object and covert it to a Python type? I am not really familiar with rubicon.objc.

Reply:
It's definitely possible to convert to a Python type... it just depends what type the object *is*. I've found a couple of references that suggest the first argument is an `NSString`; if that's the case, you should be able to do a `str()` on the argument to get a Python string. Rubicon has [docs on type conversions](https://rubicon-objc.readthedocs.io/en/latest/how-to/type-mapping.html#converting-from-objective-c-to-python) that might be helpful.

Reply:
Yes, I was somewhat aware of `py_from_ns` for conversions. I was struggling with the fact that the passed evaluation result seemed to be an `int` and not a NSString. Specifying `objc_id` as the argument type allows converting this id to a Python type later.

Regarding the return type, the Objective-c developer documentation specifies `id` as the first argument of the completion handler and the Swift documentation has the argument type `Any`. In fact, when I evaluate the javascript code `1 + 1` the result is given as a float and when evaluating `'Hello ' + 'World!'` the result is a string. The type does seem to be dynamically converted from JS to Objective-c (and then to Python).